File: LeetCode/Medium/0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (x,y) -> x[1]-y[1]);
        int ret = 0;
        int left = 0, right = 1, nextIdx = 1;
        
        while (right < intervals.length){
            if (intervals[left][1] > intervals[right][0]) ++ret;
            else left = nextIdx;
            
            ++right;
            nextIdx = right;
        }
        return ret;
    }
}

File: LeetCode/Medium/0435-non-overlapping-intervals/0435-non-overlapping-intervals.go
func eraseOverlapIntervals(intervals [][]int) int {
    sort.Slice(intervals, func (a, b int) bool {
        return intervals[a][1] < intervals[b][1]
    })
    ret := 0
    left, right, nextIdx := 0, 1, 1
    
    for right < len(intervals) {
        if intervals[left][1] > intervals[right][0] {
            ret++
        } else {
            left = nextIdx
        }
        right++
        nextIdx = right
    }
    
    return ret
    
}